src/linklabel.rs

```diff
@@ -84,13 +84,12 @@ pub(crate) fn scan_link_label_rest(text: &str) -> Option<(usize, CowStr<'_>)> {
                     }
                 }
                 if whitespaces > 1 || c != ' ' {
-                    byte_index -= c.len_utf8();
                     if still_borrowed {
-                        label.push_str(&text[start_byte..byte_index]);
+                        let end_ix = byte_index - c.len_utf8();
+                        label.push_str(&text[start_byte..end_ix]);
                         still_borrowed = false;
                     }
                     c = ' ';
-                    byte_index += c.len_utf8();
```

Comment:
I agree this change is better (less mutation), but swapping lines 92 and 93 would also have worked, yes?

Reply:
Correct, that was my initial fix ^^